### PR TITLE
feat(plugins): adding halyard commands for plugins

### PR DESCRIPTION
**Pull request by link108**
This PR is related to https://github.com/spinnaker/spinnaker/issues/4181.

This PR adds commands to Halyard to configure plugins, as well as lay down `plugins.yml` and configuration updates to `orca.yml` on deploy.
These files are used by Kork in order to enable plugins (https://github.com/spinnaker/kork/pull/325).

### CODE REVIEW

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/OrcaProfileFactory.java [hunk 1]
```diff
@@ -73,6 +79,20 @@ protected void setProfile(
...
     profile.appendContents("pipelineTemplate.enabled: " + pipelineTemplates);
+
+    final List<Plugin> plugins = deploymentConfiguration.getPlugins().getPlugins();
+    Map<String, Object> fullyRenderedYaml = new LinkedHashMap<>();
+    Map<String, Object> pluginMetadata =
+        plugins.stream()
+            .filter(p -> p.getEnabled())
+            .filter(p -> !p.getManifestLocation().isEmpty())
+            .map(p -> p.generateManifest())
+            .collect(Collectors.toConcurrentMap(m -> m.getName(), m -> m.getOptions()));
```
Comment (ezimanyi) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/OrcaProfileFactory.java]: Does this map need to be concurrent?

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/OrcaProfileFactory.java [hunk 2]
```diff
@@ -73,6 +79,20 @@ protected void setProfile(
     profile.appendContents("pipelineTemplates.enabled: " + pipelineTemplates);
     // For backward compatibility
     profile.appendContents("pipelineTemplate.enabled: " + pipelineTemplates);
+
+    final List<Plugin> plugins = deploymentConfiguration.getPlugins().getPlugins();
```
Comment (ezimanyi) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/OrcaProfileFactory.java]: Why are we directly appending to the `orca.yml` config file?  Aren't we writing out a separate `plugins.yml` file that will contain the plugin config and that each service supporting plugins will look at?

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/OrcaProfileFactory.java (hunk 1 above)
Comment (link108) [reply halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/OrcaProfileFactory.java]: Nope, was thinking of making this a parallel stream, but I think thats overkill. Moving to `Collectors.toMap` instead

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/OrcaProfileFactory.java (hunk 2 above)
Comment (link108) [reply halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/OrcaProfileFactory.java]: We are writing out a `plugins.yml` file, it is for use prior to spring context initialization and contains the plugin names and the locations of jars to download. `orca.yml` will contain the configurations for the plugins, eg `url` for a service.

**Pull request merged**